setup_miner.sh

```diff
@@ -123,17 +123,64 @@ print_status "uv version: $(uv --version)"
 print_status "Detecting external IP address..."
 EXTERNAL_IP=""
 
-# Try multiple methods to get external IP
-if command -v curl &> /dev/null; then
-    EXTERNAL_IP=$(curl -s -4 ifconfig.me || curl -s -4 ipinfo.io/ip || curl -s -4 icanhazip.com)
-elif command -v wget &> /dev/null; then
-    EXTERNAL_IP=$(wget -qO- -4 ifconfig.me || wget -qO- -4 ipinfo.io/ip)
+# Function to validate if response is a valid IP address
+is_valid_ip() {
+    local ip=$1
+    # Check if it's a valid IPv4 address (basic validation)
+    if [[ $ip =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+        return 0
+    fi
+    return 1
+}
```

Comment:
```suggestion
is_valid_ip() {
    local ip=$1
    if [[ $ip =~ ^([0-9]{1,3}\.){3}[0-9]{1,3}$ ]]; then
        local IFS='.'
        read -ra octets <<< "$ip"
        for octet in "${octets[@]}"; do
            if ((octet > 255)); then
                return 1
            fi
        done
        return 0
    fi
    return 1
}
```
should validate that octets are 0-255